### PR TITLE
3991 new module openmsidmerger

### LINKED ISSUE #3991: new module: openms/idmerger
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jonasscheid**
- Added [IDMerger](https://openms.de/doxygen/release/3.0.0/html/TOPP_IDMerger.html) from OpenMS.
- Adjusted `test_data.config` accordingly

## PR checklist

Closes #3991  

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/config/test_data.config [hunk 1]
```diff
@@ -662,8 +662,9 @@ params {
             'parameter' {
                 maxquant                        = "${params.test_data_base}/data/proteomics/parameter/mqpar.xml"
             }
-            'idfile' {
-                openms_idxml                    = "${params.test_data_base}/data/proteomics/openms_idxml/BSA_QC_file.idXML"
```
Comment (pinin4fjords) [inline tests/config/test_data.config:666]: Is removing this intentional? Aw we sure nobody is using it?

Comment (jonasscheid) [reply tests/config/test_data.config:666]: Yes! :) I asked the person, who committed this -> https://github.com/nf-core/test-datasets/commit/f33fb62c57ae463c82ad6bbea6082516e9b81a8b#commitcomment-130295954
He never uploaded it on the testdata repo

Comment (jonasscheid) [reply tests/config/test_data.config:666]: Ah yes and I double-checked that nobody is using it

**Review by pinin4fjords — APPROVED** [review]
LGTM

**Pull request merged**